calculator/src/App.js

```diff
@@ -0,0 +1,73 @@
+import { Component } from "react";
+import "./App.css";
+import Buttons from "./Components/Buttons";
+import Input from "./Components/Input";
+
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { input: "0", operator: "", previousNumber: ""};
+  }
+
+
+ 
+
+  handleInput = (e) => {
+    
+    this.setState({ input: e.target.value});
+  };
+
+   handleZero = (value) => {
+     if (this.state.input !== "") {
+       this.setState({ input: this.state.input + value})
+     }
+   }
+
+  
+
+   handleOp = (e) => {
+     this.setState({
+       operator: e.target.value, previousNumber: this.state.input })
+   }
+
+
+   handleEqual = () => { 
+     const {operator, input, previousNumber} = this.state
+     let sum = 0
```

Comment:
I don't think we need this sum variable, we can do the calc right inside setState